src/api/users/mutations/loginUser/index.js

```diff
@@ -0,0 +1,71 @@
+// @flow
+import bcrypt from 'bcrypt'
+import type { ValidationError } from '@types/ValidationError'
+import createToken from '@models/Token/createToken'
+import retrieveToken from '@models/Token/retrieveToken'
+import User from '@models/User'
+import { AUTHENTICATION_ERROR_INVALID_CREDENTIALS } from '@constants/errors'
+
+export type LoginUserArgs = {
+  email: string,
+  password: string
+}
+
+type LoginUserResponse = {
+  token?: string,
+  errors: Array<ValidationError>
+}
+
+const resolver = async (
+  _: any,
+  args: LoginUserArgs
+): Promise<LoginUserResponse> => {
+  const errors: Array<ValidationError> = []
+  const email: string = args.email
+  const password: string = args.password
+
+  if (!email) {
+    errors.push({ field: 'email', message: 'This field must not be empty' })
+  }
+
+  if (!password) {
+    errors.push({ field: 'password', message: 'This field must not be empty' })
+  }
+
+  const user: User = await User.query().findOne({ email })
+
+  if (user === undefined) {
+    errors.push({
+      field: 'generic',
+      message: AUTHENTICATION_ERROR_INVALID_CREDENTIALS
+    })
+
+    return { status: 'FAILURE', errors }
+  }
```

Comment:
- Oh. I thought it would return `null` and not `undefined` 😄 Let’s just use `if (!user)`, testing for `=== undefined` is not something you usually do.
- I’d move line 46-49 to line 36 and simply have `if (user === undedfined || isPasswordMatching)` as condition to avoid repetition. Especially since you don’t return early line 56 and still try to retrieve the token.

Reply:
Completely right here